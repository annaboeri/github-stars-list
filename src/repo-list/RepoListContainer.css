/* Repo List Styles */
.error-container, .loading-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 50vh;
  justify-content: center;
}

.cards .error-container, .cards .loading-container {
  height: 10em;
}

.repo-list-container {
  margin: 0 auto;
  padding: 0 1em;
}

@media screen and (min-width: 52em) {
  .repo-list-container {
    max-width: 75em;
 }
}

/* Card Styles */
.cards {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  margin: 20px;
}

.flip-card {
  background-color: transparent;
  height: 250px;
  margin: 1em auto;
  perspective: 1000px;
  width: 85vw;
}

.flip .card-content {
  transform: rotateY(180deg);
  -webkit-transform: rotateY(180deg); /* Ch <36, Saf 5.1+, iOS < 9.2, An =<4.4.4 */
      -ms-transform: rotateY(180deg); /* IE 9 */
          transform: rotateY(180deg); /* IE 10, Fx 16+, Op 12.1+ */
}

.card-content {
  background-color: var(--green-faint);
  border: 2px solid var(--green-light);
  border-radius: 5px;
  height: 100%;
  position: relative;
  text-align: center;
  transition: transform 0.8s;
  transform-style: preserve-3d;
  -webkit-transform-style: preserve-3d;
  width: 95%;
}

.card-content h3 {
  font-weight: 700;
  margin: .5em;
}

.card-content p {
  font-size: 80%;
  margin: 1em;
}

.flip-card-front, .flip-card-back {
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
    -moz-backface-visibility: hidden;
  height: 100%;
  position: absolute;
  width: 100%;
}

.flip-card-front {
  padding-top: 15px;
}

.flip-card-back {
  overflow-y: scroll;
  text-align: left;
  transform: rotateY(180deg);
}

.repo-link {
  font-size: 80%;
}

.repo-link:hover .fa-github {
  color: var(--green-medium);
}

/* Card Media Queries */
@media screen and (min-width: 40em) {
  .cards {
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
  }
  .flip-card {
    margin-bottom: 1em;
    width: 300px;
  }
}

@media screen and (min-width: 60em) {
  .flip-card {
    margin-bottom: 2em;
    width: 350px;
  }
}

/* Commit List Styles */
.commit-list-container {
  margin: 0 1em 2em;
  text-align: center;
}

.commit-list-container h3 {
  color: var(--green-medium);
  font-size: 90%;
}

.commit-list-container p {
  margin: 0.5em 0.25em;
  text-align: left;
}

.commit {
  padding-bottom: 0.25em;
}

.commit-message {
  font-weight: 600;
}

.commit-message:hover {
  color: var(--green-medium);
}
