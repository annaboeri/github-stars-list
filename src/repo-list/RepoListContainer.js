import React, { Component } from 'react';
import RepoCard from './RepoCard';

class RepoListContainer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      isLoaded: false,
      repos: [],
    };
    this.fetchRepos = this.fetchRepos.bind(this);
  }

  componentDidMount() {
    this.fetchRepos();
  }

  fetchRepos() {
    const query = "q=stars:>1000&sort=stars&per_page=100";
    const apiUrl = `https://api.github.com/search/repositories?${query}`;
    fetch(apiUrl)
      .then(res => res.json())
      .then(
        (result) => {
          console.log('api call result:', result)
          this.setState({
            isLoaded: true,
            repos: result.items
          });
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
          this.setState({
            isLoaded: true,
            error
          });
        }
      )
  }

  render() {
    const { error, isLoaded, repos } = this.state;
    if (error) {
      return <div>Error: {error.message}</div>;
    } else if (!isLoaded) {
      return <div>Loading...</div>;
    } else {
      return (
        <section className="repo-list-container">
          <ul>
            {repos.map(repo =>
              <RepoCard repo={repo} key={repo.id} />
            )}
          </ul>
        </section>
      );
    }
  }
}

export default RepoListContainer;
