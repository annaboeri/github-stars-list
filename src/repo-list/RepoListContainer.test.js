import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RepoListContainer from './RepoListContainer';

describe('RepoListContainer', () => {
  it('should render loading state', (done) => {
    const component = shallow(<RepoListContainer />);
    const tree = toJson(component);

    expect(tree).toMatchSnapshot();
    expect(component.find('.loading-container').exists()).toBe(true);
    done();
  });

  it('should render repos', (done) => {
    const component = shallow(<RepoListContainer />);
    const tree = toJson(component);
    const mockState = { isLoaded: true }
    component.setState({ ...mockState });

    expect(tree).toMatchSnapshot();
    expect(component.find('.repo-list-container').exists()).toBe(true);
    done();
  });

  it('should render error state', (done) => {
    const component = shallow(<RepoListContainer />);
    const tree = toJson(component);
    const mockState = { error: true }
    component.setState({ ...mockState });

    expect(tree).toMatchSnapshot();
    expect(component.find('.error-container').exists()).toBe(true);
    done();
  });
});
