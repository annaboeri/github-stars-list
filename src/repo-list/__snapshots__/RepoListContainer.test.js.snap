// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoListContainer should render error state 1`] = `
<section
  className="loading-container"
>
  <div
    className="fa-3x"
  >
    <i
      className="fas fa-spinner fa-pulse"
    />
  </div>
  <p>
    Loading...
  </p>
</section>
`;

exports[`RepoListContainer should render loading state 1`] = `
<section
  className="loading-container"
>
  <div
    className="fa-3x"
  >
    <i
      className="fas fa-spinner fa-pulse"
    />
  </div>
  <p>
    Loading...
  </p>
</section>
`;

exports[`RepoListContainer should render repos 1`] = `
<section
  className="loading-container"
>
  <div
    className="fa-3x"
  >
    <i
      className="fas fa-spinner fa-pulse"
    />
  </div>
  <p>
    Loading...
  </p>
</section>
`;
